src/07-embedding/02-artists.ts

```diff
@@ -0,0 +1,151 @@
+import { ARTIST_IDS } from "./artist-ids"
+import weaviate from "weaviate-ts-client"
+import _ from "lodash"
+
+const client = weaviate.client({
+  scheme: "https",
+  host: "https://weaviate.stg.artsy.systems",
+})
+
+const CLASS_NAME: string = "ArtistBio2"
+
+const BATCH_SIZE = 100
+
+async function main() {
+  const artists = await fetchArtists()
+  await prepareArtistsCollection()
+  await insertArtists(artists)
+  /*
+   * now query from Weaviate console:
+   * https://link.weaviate.io/3UEZ854
+   */
+}
+
+main()
+
+async function prepareArtistsCollection() {
+  const client = weaviate.client({
+    scheme: "https",
+    host: "https://weaviate.stg.artsy.systems",
+  })
+
+  const alreadyExists = await client.schema.exists(CLASS_NAME)
+
+  if (alreadyExists) {
+    console.log(`${CLASS_NAME} class already exists, deleting it`)
+    await client.schema.classDeleter().withClassName(CLASS_NAME).do()
+  }
+
+  const classWithProps = {
+    class: CLASS_NAME,
+    vectorizer: "text2vec-openai",
+    moduleConfig: {
+      "text2vec-openai": {
+        model: "ada",
+        dimensions: 1536,
+        type: "text",
+        vectorizeClassName: false,
+      },
+    },
+    properties: [
+      {
+        name: "blurb",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: false,
+            vectorizePropertyName: false,
+          },
+        },
+      },
+      {
+        name: "slug",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+          },
+        },
+      },
+    ],
+  }
+
+  const classResult = await client.schema
+    .classCreator()
+    .withClass(classWithProps)
+    .do()
+
+  console.log(JSON.stringify(classResult, null, 2))
+}
+
+async function insertArtists(artists: { slug: string; blurb: string }[]) {
+  console.log(`Going to insert ${artists.length} artists`)
+
+  const batches = _.chunk(artists, BATCH_SIZE)
+  console.log(`Inserting ${batches.length} batches`)
+
+  for (const artistBatch of batches) {
+    let batcher = client.batch.objectsBatcher()
+    batcher = batcher.withObjects(
+      ...artistBatch.map((artist) => ({
+        class: CLASS_NAME,
+        properties: artist,
+      }))
+    )
+    process.stdout.write(".")
+    await batcher.do()
+  }
+  process.stdout.write("\n")
+}
```

Comment:
We make use of Weaviate's batching API, which does the OpenAI vectorizing, as well as the db insertion, in batches for better performance.